airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -191,18 +189,25 @@ def close_partition(self, partition: Partition) -> None:
         self._has_closed_at_least_one_slice = True
 
     def _add_slice_to_state(self, partition: Partition) -> None:
+        most_recent_cursor_value = self._most_recent_cursor_value_per_partition.get(partition)
+
         if self._slice_boundary_fields:
             if "slices" not in self.state:
                 raise RuntimeError(
                     f"The state for stream {self._stream_name} should have at least one slice to delineate the sync start time, but no slices are present. This is unexpected. Please contact Support."
                 )
             self.state["slices"].append(
                 {
-                    "start": self._extract_from_slice(partition, self._slice_boundary_fields[self._START_BOUNDARY]),
-                    "end": self._extract_from_slice(partition, self._slice_boundary_fields[self._END_BOUNDARY]),
+                    self._connector_state_converter.START_KEY: self._extract_from_slice(
```

Comment:
Why have we changed `start` to `self._connector_state_converter.START_KEY`? Are we afraid that this might be a change for some connectors where `self._connector_state_converter.START_KEY` would be different than `start`?

Reply:
I believe it would be better to have an interface to change key boundary names with a state converter, similar to the approach we use when there are no slice boundary fields and the `most_recent_cursor_value` is present.